packages/material-ui-lab/src/MasonryItem/MasonryItem.js

```diff
@@ -0,0 +1,106 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import clsx from 'clsx';
+import { unstable_composeClasses as composeClasses } from '@material-ui/unstyled';
+import { styled, useThemeProps } from '@material-ui/core/styles';
+import { getMasonryItemUtilityClass } from './masonryItemClasses';
+import MasonryContext from '../Masonry/MasonryContext';
+
+const useUtilityClasses = (styleProps) => {
+  const { classes } = styleProps;
+
+  const slots = {
+    root: ['root'],
+  };
+
+  return composeClasses(slots, getMasonryItemUtilityClass, classes);
+};
+
+const MasonryItemRoot = styled('div', {
+  name: 'MuiMasonryItem',
+  slot: 'Root',
+  overridesResolver: (props, styles) => {
+    return [styles.root];
+  },
+})(({ styleProps, theme }) => {
+  const gap = Number(theme.spacing(styleProps.spacing).replace('px', ''));
+  const rowSpan = styleProps.height ? Math.ceil(styleProps.height + gap) : 0;
+  return {
+    width: '100%',
+    [`& *`]: {
```

Comment:
Should we maybe select here first child only? If the content is more complex tree, this may create problems.

Reply:
What I mean here, was trying to use:
[`& > *`]: {